lib/concentrate/pipeline.ex

```diff
@@ -0,0 +1,41 @@
+defmodule Concentrate.Pipeline do
+  @moduledoc """
+  Behavior for defining a concentrate pipeline. Generally these included sources and consumers, and sometimes also a merge step.
+  """
+
+  @callback init(keyword()) :: [Supervisor.child_spec()]
+
+  @spec start_link(module(), keyword()) :: Supervisor.on_start()
+  def start_link(module, opts) do
+    children = module.init(opts)
+
+    Supervisor.start_link(children, strategy: :rest_for_one)
+  end
+
+  def child_spec(opts) do
```

Comment:
Does this need a `@impl true` to say it implements the `Supervisor.child_spec` callback? (Or would that get in the way of this being a behavior itself?)

(Same for `Concentrate.Supervisor.child_spec` which doesn't currently have an `@impl true`.)

Reply:
I don't believe so because these aren't declaring a behaviour. I added a typespec here though.